source/initial_composition/slab_model.cc

```diff
@@ -0,0 +1,123 @@
+
+      const double depth                     = this->get_geometry_model().depth(position);
+      double slab_composition                = 0;
+
+      // The first column after the points correspond to the slabs depth and the second to the slabs thickness
+      double slab_depths    = slab_boundary.get_data_component(surface_boundary_id, position, 0);
+      double slab_thickness = slab_boundary.get_data_component(surface_boundary_id, position, 1);
+
+      // The input ascii file is structure and as a result has many data points where slabs are absent. We give a very high
+      // number at those locations and therefore the first two conditions check if we are within the slabs.
+      // In the input file, slab depths are to the top of the slab surface.
+      // The hyperbolic tangent function smooths the slabs in the depth direction.
+      if ( (compositional_index == slab_index) && (slab_depths < 1e10) && (slab_thickness < 1e10) &&
+           (depth >= slab_depths) && (depth <= slab_depths + slab_thickness) )
+        {
+          const double slab_center    = slab_depths + slab_thickness/2.;
+          const double half_thickness = slab_thickness/2;
+          slab_composition            = ( 1 - std::tanh( 10 * (std::abs (slab_center - depth) - half_thickness)/half_thickness ) )/2;
+        }
+
+      return slab_composition;
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        Utilities::AsciiDataBoundary<dim>::declare_parameters(prm,  "../../input_data/", "slab2_depth_thickness_2D.txt", "Slab model");
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        slab_boundary.initialize_simulator (this->get_simulator());
+        slab_boundary.parse_parameters(prm, "Slab model");
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(SlabModel,
+                                              "slab model",
+                                              "An initial composition model computed to define slabs "
+                                              "using the slab2 model (Hayes et al., 2018) input in "
+                                              "an ascii format. The file describes the depths to the "
+                                              "top of the slabs and their thickness."
+                                              "The computed compositional value is 1 within the slabs "
+                                              "and zero elsewhere. In order to prevent sharp jumps, the "
+                                              "slabs are smoothened in the depth direction. "
```

Comment:
Either adjust or remove.

```suggestion
                                              "and zero elsewhere. "
```